### PR TITLE
Datastore/pagination

### PR DESCRIPTION
**Pull request by alexscoelho**
Datastore pagination and github username

### CODE REVIEW

File: breathecode/activity/views.py [hunk 1]
```diff
@@ -203,6 +203,14 @@ def get(self, request, cohort_id=None, academy_id=None):
 
         datastore = Datastore()
         #academy_iter = datastore.fetch(**kwargs, academy_id=int(academy_id))
+        limit = request.GET.get('limit')
```
Comment (jefer94) [inline breathecode/activity/views.py:206]: Add this pagination to `ActivityMeView` too

File: breathecode/activity/tests/urls/tests_cohort.py [hunk 2]
```diff
@@ -0,0 +1,83 @@
...
+
+class MediaTestSuite(MediaTestCase):
+    """
+    🔽🔽🔽 With data
+    """
+    @patch.object(Datastore, '__init__', new=lambda x: None)
+    @patch.object(Datastore,
+                  'fetch',
+                  new=datastore_fetch_mock(first_fetch=DATASTORE_SEED))
+    def test_get_activities(self):
```
Comment (jefer94) [inline breathecode/activity/tests/urls/tests_cohort.py:46]: We need a test:

1. With more that 100 activities without passing paginations that just get 100 elements
2. With 20 elements, the `limit` could be 10, the `offset` could be 0, this returns the first 10 elements
3. With 20 elements, the `limit` could be 10, the `offset` could be 10, this returns the last 10 elements
4. With 20 elements, the `limit` could be 10, the `offset` could be 20, this returns zero elements

This test should copy and paste to `activity/me` and remember you should use a generator to generate a list with 10 or 100 elements of same dictionary

**Review by jefer94 — CHANGES_REQUESTED** [review]

File: breathecode/activity/actions.py [hunk 3]
```diff
@@ -0,0 +1,4 @@
+def paginated_datastore(public_iter):
```
Comment (jefer94) [inline breathecode/activity/actions.py:1]: this is being used?

File: breathecode/activity/tests/urls/tests_cohort.py [hunk 4]
```diff
@@ -0,0 +1,295 @@
...
+
+    """
+    🔽🔽🔽 Without pagination
+    """
+
+    @patch.object(Datastore, '__init__', new=lambda x: None)
+    @patch.object(Datastore,
+                  'fetch',
+                  new=datastore_fetch_mock(first_fetch=generate_data(100)))
+    @patch.object(Datastore, 'count', new=datastore_count_mock(7957599))
```
Comment (jefer94) [inline breathecode/activity/tests/urls/tests_cohort.py:157]: You should prevent call `.count` and using this mock if `limit` or `offset` is not set

File: breathecode/activity/tests/urls/tests_cohort.py [hunk 5]
```diff
@@ -0,0 +1,295 @@
...
+                             role='potato',
+                             cohort_kwargs=cohort_kwargs)
+
+        url = reverse_lazy('activity:academy_cohort_id',
+                           kwargs={'cohort_id': 1})
+        response = self.client.get(url)
+
+        json = response.json()
+
+        self.assertEqual(len(json), 100)
```
Comment (jefer94) [inline breathecode/activity/tests/urls/tests_cohort.py:176]: This assert need be more exact

File: breathecode/activity/tests/urls/tests_cohort.py [hunk 6]
```diff
@@ -0,0 +1,295 @@
...
+        self.assertEqual(response.status_code, status.HTTP_200_OK)
+
+        self.assertEqual(mock.fetch.call_args_list, [
```
Comment (jefer94) [inline breathecode/activity/tests/urls/tests_cohort.py:179]: Missing the asserts to `count` function

File: breathecode/activity/tests/urls/tests_cohort.py [hunk 7]
```diff
@@ -0,0 +1,295 @@
...
+                             capability='classroom_activity',
+                             role='potato',
+                             cohort_kwargs=cohort_kwargs)
+
+        url = reverse_lazy('activity:academy_cohort_id',
+                           kwargs={'cohort_id': 1}) + '?limit=10'
+        response = self.client.get(url)
+
+        json = response.json()
+        results = len(json['results'])
```
Comment (jefer94) [inline breathecode/activity/tests/urls/tests_cohort.py:212]: This test need perform a test about the wrapper of pagination

**Review by jefer94 — CHANGES_REQUESTED** [review]

**Review by alexscoelho — COMMENTED** [review]
Fixed comments!

**Review by jefer94 — APPROVED** [review]

**Pull request merged**